test/Chainweb/Test/Rosetta/RestAPI.hs

```diff
@@ -209,49 +230,136 @@ blockTests tio envIo = testCaseSchSteps "Block Tests" $ \step -> do
+        Just rem1 -> validateOp 0 "TransferOrCreateAcct" e7f7ks Remediation (negate 100) rem1
+        Nothing -> pure ()
+
+      -- rest txs (i.e. transfer transaction)
+      validateOp 0 "FundTx" sender00ks Successful (negate defFundGas) fundtx
+      validateOp 1 "TransferOrCreateAcct" sender01ks Successful 1.0 cred
+      validateOp 2 "TransferOrCreateAcct" sender00ks Successful (negate 1.0) deb
+      validateOp 3 "GasPayment" sender00ks Successful (defFundGas - transferGasCost) redeem
+      validateOp 4 "GasPayment" noMinerks Successful transferGasCost reward
+
+blockCoinV2RemediationTests :: RosettaTest
+blockCoinV2RemediationTests _ envIo =
+  testCaseSchSteps "Block CoinV2 Remediation Tests" $ \step -> do
+    cenv <- envIo
 
+    step "fetch coin v2 remediation block"
+    resp <- block cenv (req bhCoinV2Rem)
+
+    step "validate block"
+    _blockResp_otherTransactions resp @?= Nothing
+    Just b <- pure $ (_blockResp_block resp)
+    _block_metadata b @?= Nothing
+    _blockId_index (_block_blockId b) @?= bhCoinV2Rem
+    _blockId_index (_block_parentBlockId b) @?= (bhCoinV2Rem - 1)
+
+    case (_block_transactions b) of
+      x:y:z:_ -> do
+        step "check remediation transactions' request keys"
+        -- TODO: are these unique across lifetime of blockchain/chains?
+        [ycmd, zcmd] <- upgradeTransactions v cid
+        _transaction_transactionId y @?= (pactHashToTransactionId (_cmdHash ycmd))
+        _transaction_transactionId z @?= (pactHashToTransactionId (_cmdHash zcmd))
+
+        step "check remediation transactions' operations"
+        _transaction_operations y @?= [] -- didn't touch the coin table
+        _transaction_operations z @?= [] -- didn't touch the coin table
+                                         -- NOTE: no remedition withdrawl happens in this version
+
+        step "check coinbase transaction"
+        [cbase] <- pure $ _transaction_operations x
+        validateOp 0 "CoinbaseReward" noMinerks Successful defMiningReward cbase
+
+      _ -> assertFailure $ "coin v2 remediation block should have at least 3 transactions:"
+           ++ " coinbase + 2 remediations"
+  where
+    bhCoinV2Rem = 1
+    req h = BlockReq nid $ PartialBlockId (Just h) Nothing
+
+block20ChainRemediationTests :: RosettaTest
+block20ChainRemediationTests _ envIo =
+  testCaseSchSteps "Block 20 Chain Remediation Tests" $ \step -> do
+    cenv <- envIo
+
+    step "fetch  remediation block"
+    resp <- block cenv (req bhChain20Rem)
+
+    step "validate block"
+    _blockResp_otherTransactions resp @?= Nothing
+    Just b <- pure $ (_blockResp_block resp)
+    _block_metadata b @?= Nothing
+    _blockId_index (_block_blockId b) @?= bhChain20Rem
+    _blockId_index (_block_parentBlockId b) @?= (bhChain20Rem - 1)
+
+    case (_block_transactions b) of
+      x:y:_ -> do
+        step "check remediation transactions' request keys"
+        -- TODO: are these unique across lifetime of blockchain/chains?
```

Comment:
@emilypi is this correct? Are the request keys for the remediations unique across the lifetime of the blockchain and across chains? Unique transaction id is required for rosetta and usually that's not a problem but remediations command payloads are not included into the block like coinbase/user transactions are. So I wanted to make sure if this assumption was correct?

Reply:
I'm pretty sure that request keys for remediations are fixed across all chains, yes. Otherwise, we'd not be able to replay them. They do go into the hash of the block at that height iirc.

Reply:
I thought only the logs of the remediations went into the hash?
I guess my question was is it possible that we could get a user-generated transaction that has the same request key as one of the remediations?

Reply:
I'm not sure if it's only the logs, since we're calling the actual pact code and affecting the pact db environment. It may live exclusively in there, but I'd have to double check. As far as request keys... no. The request keys are generated by a hash of the transaction data and a nonce, and we track nonces to ensure they're unique - so request keys can be considered unique